src/anm/animation/animation.js

```diff
@@ -166,11 +167,34 @@ Animation.prototype.traverse = function(visitor, data) {
  *
  * @param {Function} visitor
  * @param {anm.Element} visitor.child
+ * @param {Boolean} visitor.return if `false` returned, stops the iteration. no-`return` or empty `return` both considered `true`.
  * @param {Object} [data]
  */
 Animation.prototype.each = function(visitor, data) {
     for (var i = 0, tlen = this.tree.length; i < tlen; i++) {
-        visitor(this.tree[i], data);
+        if (visitor(this.tree[i], data) === false) break;
```

Comment:
You need to write test test that makes sure the element is found, this is not some abstract “testing users’ code”
As long as you have tests for this code I’m fine even with returning false, it’s less evil than having no tests *especially* in dynamic languages

> On May 26, 2015, at 8:39 AM, Ulric Wilfred <notifications@github.com> wrote:
> 
> In src/anm/animation/animation.js <https://github.com/Animatron/player/pull/306#discussion_r31028390>:
> 
> >   * @param {Object} [data]
> >   */
> >  Animation.prototype.each = function(visitor, data) {
> >      for (var i = 0, tlen = this.tree.length; i < tlen; i++) {
> > -        visitor(this.tree[i], data);
> > +        if (visitor(this.tree[i], data) === false) break;
> Tests are not the universal pill. Especially in JS. Sometimes you just need to trust yourself.
> 
> Again, it's a matter of decision, not a test. Returning false to skip element is quite common practice for JS, also it's properly noticed in docs. And, it's common to omit return in dynamic languages. You can't test users' code, who usually omit/forget return (like in 90% cases), especially while using .each.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Animatron/player/pull/306/files#r31028390>.
>